verify/assets/index.html

```diff
@@ -116,153 +155,161 @@ <h2>Twilio Verify</h2>
-            $("#verification_code").val("");
-          } else {
-            showVerificationStatus("danger", json.message);
-            $("#verification_code").val("");
-          }
-        })
-        .catch(err => {
-          console.log(err);
-          showVerificationStatus("danger", "Something went wrong!");
-          $("#verification_code").val("");
+      try {
+        const response = await fetch("./check-verify", {
+          method: "POST",
+          body: data,
         });
-    });
 
-    const locales = [
-      { text: "English", value: "en" },
-      { text: "Afrikaans", value: "af" },
-      { text: "Arabic", value: "ar" },
-      { text: "Catalan", value: "ca" },
-      { text: "Chinese", value: "zh" },
-      { text: "Chinese (Mandarin)", value: "zh-CN" },
-      { text: "Chinese (Cantonese)", value: "zh-HK" },
-      { text: "Croatian", value: "hr" },
-      { text: "Czech", value: "cs" },
-      { text: "Danish", value: "da" },
-      { text: "Dutch", value: "nl" },
-      { text: "English (British)", value: "en-GB" },
-      { text: "Finnish", value: "fi" },
-      { text: "French", value: "fr" },
-      { text: "German", value: "de" },
-      { text: "Greek", value: "el" },
-      { text: "Hebrew", value: "he" },
-      { text: "Hindi", value: "hi" },
-      { text: "Hungarian", value: "hu" },
-      { text: "Indonesian", value: "id" },
-      { text: "Italian", value: "it" },
-      { text: "Japanese", value: "ja" },
-      { text: "Korean", value: "ko" },
-      { text: "Malay", value: "ms" },
-      { text: "Norwegian", value: "nb" },
-      { text: "Polish", value: "pl" },
-      { text: "Portuguese - Brazil", value: "pt-BR" },
-      { text: "Portuguese", value: "pt" },
-      { text: "Romanian", value: "ro" },
-      { text: "Russian", value: "ru" },
-      { text: "Spanish", value: "es" },
-      { text: "Swedish", value: "sv" },
-      { text: "Tagalog", value: "tl" },
-      { text: "Thai", value: "th" },
-      { text: "Turkish", value: "tr" },
-      { text: "Vietnamese", value: "vi" }
-    ];
+        const json = await response.json();
+
+        if (json.success) {
+          showModalStatus("Verification success!", (color = "green"));
+          document.getElementById("code").value = "";
+        } else {
+          showModalStatus("Incorrect token!", (color = "#a94442"));
+          document.getElementById("code").value = "";
+        }
+      } catch (error) {
+        console.error(error);
+        showModalStatus("Something went wrong!");
+        document.getElementById("code").value = "";
+      }
+    }
 
-    const selectLocale = $("#select-locale");
+    document
+      .getElementById("check-code")
+      .addEventListener("submit", (event) => checkOtp(event));
 
-    locales.forEach(l => {
-      selectLocale.append($("<option>")
-        .val(l.value)
-        .text(l.text))
+    document.getElementById("close").addEventListener("click", () => {
+      modal.style.display = "none";
```

Comment:
Should you also run `clearStatus()` here, as below? If you don't and the user closes the modal with the cross, then the next time it opens it still has the previous success/failure message.